### ISSUE TITLE
Release 0.32

### ISSUE THREAD

**Issue by garethrees** [mysociety/alaveteli#4615]
Working on the assumption that we won’t roll out pricing in this release
* That means, any new pro features should work without needing to inspect subscriptions or rely on webhooks, or any other Stripe interaction.

What do we want to do about batch for this release?
* Gut feel is that its too early to make it available for reuse

## Before release candidate date

- [x] List of all substantial features https://github.com/mysociety/alaveteli/issues/4833 https://gist.github.com/lizconlan/b7da0839ff759164f90e26233f1ba55e
- [x] Figure out how each feature aligns with current re-user expectations of Pro
  * Do things work as expected with Pro turned off?
  * Do things work as expected with Pro turned on, but pricing disabled?
  * Check rate limit upsell – does it make sense if batch turned off?
  * Do they work without Stripe integration?
  * Do we need to make any modifications so that we can make release?

- [x] Update upgrade notes
  - [x] Run diff on `general.yml-example` to look for new/renamed keys
  - [x] Move the upgrade steps to the top of the section, in exec order (harder to miss something, plus may not be immediately obvious to migrate the db before running the timestamps script for example)

- [x] Other steps from [release manager's checklist](https://github.com/mysociety/alaveteli/wiki/Release-Manager's-checklist#before-the-release-candidate-date)
  - [x] model annotations are up to date
  - [x] update commonlib
  - [x] no deprecations for this version
  - [x] upgrade steps include `db:migrate`
  - [x] changelog - highlighted features needs updating
  - [x] changelog - template changes need to be added (may be subject to change by outstanding tickets)
  - [x] check for new translations - yes, there will be changes (for pro and core translations)
  - [x] make announcement for translators
  - [x] Update newrelic gem #4862
  - [x] script/test-vagrant-provisioning (Note: check that _this script_ works since we've dropped a few supported OS' recently)
- [x] Bug Fixes
  - [x] Requires #4858 
  - [x] Requires #4814 
  - [x] Requires #4859 
- [x] test/update supported themes (+ change specs from `FactoryGirl` to `FactoryBot`)

## Pre-release steps
- [x] Brakeman warning fix 
- [x] merge the header changes to dependent themes
- [x] work through the ["On the release date" steps of the checklist](https://github.com/mysociety/alaveteli/wiki/Release-Manager's-checklist#on-the-release-date)

**Comment by garethrees** [comment]
Be nice in https://github.com/mysociety/alaveteli/pull/4466/ if we fell back to the normal opengraph logo if one wasn't configured, because it's inevitably going to get forgotten.

**Comment by garethrees** [comment]
Test plan for "Belgium" context of non-self-serve pro:

- Sign up for a standard user account
- As an admin, check the "Pro" role on the new user's account
- Sign in as the now-pro user
- Should be able to:
  - Make a public request
  - Make a private request
  - Change an embargo on a request
  - Will not have access to any batch interfaces
  - Will not be able to go to `/profile/subscriptions` (which requires Stripe)

**Comment by garethrees** [comment]
Also do a quick smoke test of all new features with a non-WDTK theme.

**Comment by lizconlan** [comment]
It looks like the Upgrade Notes describe the config changes (aside from the new not-recommended-for-reuse keys i.e. `PRO_REFERRAL` and `PRO_BATCH_AUTHORITY_LIMIT`), although the new `BLOCK_SPAM_SIGNINS` option is not mentioned.

A preliminary smoke test upgrade from 0.31.0.3 to develop (64ac2fcef) following the upgrade steps (without a theme installed) seems fine.

**Comment by lizconlan** [comment]
#4814 would be nice to have to round out the `foi_no` tag feature

**Comment by lizconlan** [comment]
Requires #4853

**Comment by lizconlan** [comment]
See [Upgrade / Smoke tests](https://gist.github.com/lizconlan/67fffeea5991b8bea5f3efc539a14e7e) plan (run successfully)

**Comment by lizconlan** [comment]
- [ ] Need to fix this before we can recommend the current release code for reuse https://github.com/mysociety/alaveteli/issues/4858

**Comment by garethrees** [comment]
- [ ] Requires https://github.com/mysociety/alaveteli/pull/4830

**Comment by lizconlan** [comment]
May also need #4862 as newrelic_rpm is now at 5.4.0.347

**Comment by lizconlan** [comment]
Done bar the feature branch cleanup

**Comment by lizconlan** [comment]
Done! 🎉
